ext/standard/crc32.c

```diff
@@ -35,6 +39,56 @@ static unsigned long getauxval(unsigned long key) {
 		return 0;
 	return ret;
 }
+# elif defined(PHP_WIN32)
+/*
+get crc32 support info
+this is undocumented, to be confirmed
+read id_aa64isar0_el1 from registry HKLM\HARDWARE\DESCRIPTION\System\CentralProcessor\x\CP 4030
+*/
+int winaa64_support_crc32c() {
+	const WCHAR cpus_key[] = L"HARDWARE\\DESCRIPTION\\System\\CentralProcessor\\";
+	HKEY cpus_handle;
+	if (ERROR_SUCCESS != RegOpenKeyExW(HKEY_LOCAL_MACHINE, cpus_key, 0, KEY_ENUMERATE_SUB_KEYS, &cpus_handle)) {
+		return 0;
+	}
+	DWORD cpu_index = 0;
+	WCHAR cpu_num[256] = { 0 };
+	uint64_t CP4030 = 0;
+	WCHAR cpu_key[(sizeof(cpus_key) / sizeof(*cpus_key)) + 256] = L"HARDWARE\\DESCRIPTION\\System\\CentralProcessor\\";
+	while (ERROR_SUCCESS == RegEnumKeyW(cpus_handle, cpu_index++, cpu_num, 255)) {
+		if (0 != wcscpy_s(&cpu_key[sizeof(cpus_key) / sizeof(*cpus_key) - 1], 256, cpu_num)) {
+			break;
+		}
+		HKEY cpu_handle;
+		if (ERROR_SUCCESS != RegOpenKeyExW(HKEY_LOCAL_MACHINE, cpu_key, 0, KEY_READ, &cpu_handle)) {
+			break;
+		}
+		DWORD value_type;
+		uint64_t value;
+		DWORD value_size = sizeof(value);
+		if (ERROR_SUCCESS != RegQueryValueExW(cpu_handle, L"CP 4030", NULL, &value_type, (LPBYTE)&value, &value_size)) {
+			RegCloseKey(cpu_handle);
+			break;
+		}
+
+		if (REG_QWORD != value_type) {
+			RegCloseKey(cpu_handle);
+			break;
```

Comment:
Should it `return 0;` (not supported) instead of breaking if it fails to read the registry key anywhere before `CP4030 |= ~value;` for any CPU? 

Before this PR, it would have been 0 (not supported).

I'm not familiar with the registry code, though

Reply:
This is for `RegCloseKey` out of this loop, maybe using goto is better for understanding.